docs/RFCS/schema_gossip.md

```diff
@@ -0,0 +1,39 @@
+- Feature Name: schema_distribution
+- Start Date: 2015-07-20
+- RFC PR:
+- Cockroach Issue:
+
+# Summary
+
+This RFC suggests implementing eventually-consistent replication of the SQL schema to all Cockroach nodes in a cluster using gossip.
+
+# Motivation
+
+In order to support performant SQL queries, each gateway node must be able to address the data requested by the query. Today this requires the node to read `TableDescriptor`s from the KV map, which we believe (though we haven't measured) will cause a substantial performance hit which we can mitigate by actively gossiping the necessary metadata.
+
+# Detailed design
+
+Distribution of the schema metadata will be modeled after the current configuration distribution mechanism. Whenever a new `TableDescriptor` is written to the KV map, the node holding the leader lease for that key's range will add the `TableDescriptor` to gossip under a key derived from the `TableDescriptor`'s `Name` with a TTL TBD.
+
+Complete propagation of the new `TableDescriptor` will take at most `numHops * gossipInterval` where `numHops` is the maximum number of hops between any node and the publishing node, and `gossipInterval` is the maximum interval between sequential writes to the gossip network on a given node.
+
+On the read side, `sql.Server.getTableDesc`'s behaviour will change such that it reads from gossip rather than the KV store. This will require plumbing a reference to the `Gossip` instance down to the `sql.Server` instance.
```

Comment:
Perhaps it would be better to provide a function or some other mechanism to the sql.Server instance at instantiation which can fetch the latest schema for a table by name. No sense in contaminating the sql stuff with direct access to gossip.